zeq/subscriber.h

```diff
@@ -27,27 +37,77 @@ class Subscriber : public Receiver
 {
 public:
     /**
-     * Create and subscribe to one or more publishers.
+     * Create a default subscriber.
      *
-     * A subscription to a non-existing publisher is valid. It will start
-     * receiving events once the other publisher(s) is(are) publishing.
+     * Constrains:
+     * - discovers publishers on _zeroeq_pub._tcp ZeroConf service
+     * - filters session \<username\> or ZEROEQ_SESSION from environment if set
      *
-     * @param uri publishing URI in the format scheme://[*|host|IP|IF][:port]
-     * @throw std::runtime_error when the subscription failed.
+     * @throw std::runtime_error if ZeroConf is not available
      */
-    ZEQ_API explicit Subscriber( const servus::URI& uri );
+    ZEQ_API Subscriber();
+
+    /**
+     * Create a subscriber which subscribes to publisher(s) on the given URI.
+     *
+     * Constrains:
+     * - discovers publishers on _zeroeq_pub._tcp ZeroConf service if URI is not
+     *   fully qualified
+     * - filters session \<username\> or ZEROEQ_SESSION from environment if set
+     *
+     * @param uri publisher URI in the format [scheme://][*|host|IP|IF][:port]
+     * @throw std::runtime_error if ZeroConf is not available and URI is not
+     *                           fully qualified
+     */
+    ZEQ_API explicit Subscriber( const URI& uri );
+
+    /**
+     * Create a subscriber which subscribes to publisher(s) the from given
+     * session.
+     *
+     * Constrains:
+     * - discovers publishers on _zeroeq_pub._tcp ZeroConf service
+     * - filters for given session or ZEROEQ_SESSION from environment if set
```

Comment:
For symmetry to the publisher the session argument should take precedence over ZEROEQ_SESSION and an empty string should throw.